File: src/036/p036.rs
fn palindrome(n: String) -> bool {
    let reversed = n.chars().rev().collect::<String>();
    reversed == n.to_string()
}

fn number_palindrome(n: i32, base: i8) -> bool {
    match base {
        2 => palindrome(format!("{:b}", n)),
        10 => palindrome(n.to_string()),
        _ => false,
    }
}

fn double_base_palindrome(n: i32) -> bool {
    number_palindrome(n, 10) && number_palindrome(n, 2)
}

fn sum_double_base_palindromes_below(limit: i32) -> i32 {
    (0..limit)
        .into_iter()
        .filter(|&x| double_base_palindrome(x))
        .sum()
}

fn solve() -> i32 {
    sum_double_base_palindromes_below(1000000)
}

fn main() {
    let result = solve();
    println!("{}", result);
}

File: src/036/p036.swift
extension String {
    var palindrome: Bool {
        return String(self.reversed()) == self
    }
}

extension Int {
    func palindrome(base: Int) -> Bool {
        switch base {
        case 2: return String(self, radix: 2).palindrome
        case 10: return String(self).palindrome
        default: return false    
        }
    }

    func doubleBasePalindrome() -> Bool {
        return self.palindrome(base: 10) && self.palindrome(base: 2)
    }
}

func sumDoublePalindromeBelow(limit: Int) -> Int {
    return (1...limit).filter { $0.doubleBasePalindrome() }.reduce(0, +)
}


func solve() -> Int {
    return sumDoublePalindromeBelow(limit: 1000000)
}

let result = solve()
print(result)
